### PR TITLE
ARROW-3925: [Python] Add autoconf to conda install instructions

### PR DESCRIPTION
**Pull request by fjetter**
autoconf is required by the jemalloc build

### CODE REVIEW

**Comment by kszucs** [comment]
On long term we might want to prefer:

```bash
conda create -y -n pyarrow-dev -c conda-forge
    --file arrow/ci/conda_env_cpp.yml
    --file arrow/ci/conda_env_python.yml
```

which will be tested by the docker containers.

File: python/doc/source/development.rst [hunk 1]
```diff
@@ -81,7 +81,8 @@ from conda-forge:
    conda create -y -q -n pyarrow-dev \
          python=3.6 numpy six setuptools cython pandas pytest \
          cmake flatbuffers rapidjson boost-cpp thrift-cpp snappy zlib \
-         gflags brotli jemalloc lz4-c zstd -c conda-forge
+         gflags brotli jemalloc lz4-c zstd autoconf glog \
```
Comment (kszucs) [inline python/doc/source/development.rst]: `jemalloc` is not needed anymore

**Comment by fjetter** [comment]
There are cases for packages which I would like to install via conda, e.g. `autoconf` which are installed by the docker containers using apt-get. What should our approach be in these cases?
1. Add autoconf to the `conda_env_cpp.yml` and install it twice in the docker container?
2. Add autoconf to the `conda_env_cpp.yml` but don't install it using apt-get?
3. Don't put it in the file but on the cmd line, e.g. `conda create -n pyarrow-dev autoconf --file ...`?

@kszucs Do you have any preferences?

**Comment by kszucs** [comment]
@fjetter nice summary :) Go with the second.

**Comment by wesm** [comment]
@fjetter ping on this

**Comment by kszucs** [comment]
@wesm I'll finalize this PR.

What do You think about using 

```bash
conda create -y -n pyarrow-dev -c conda-forge
    --file arrow/ci/conda_env_cpp.yml
    --file arrow/ci/conda_env_python.yml
    python=3.6
```

in the development documentation?

**Comment by wesm** [comment]
That's fine with me. Fewer things to maintain

**Comment by wesm** [comment]
May have to use backslashes in the Windows version :)

**Comment by wesm** [comment]
Pinging about this

**Comment by kszucs** [comment]
@wesm pong

File: python/doc/source/development.rst [hunk 2]
```diff
@@ -76,13 +76,23 @@ Using Conda
...
 .. code-block:: shell
 
-   conda create -y -q -n pyarrow-dev \
-         python=3.6 numpy six setuptools cython pandas pytest \
-         cmake flatbuffers rapidjson boost-cpp thrift-cpp snappy zlib \
-         gflags brotli jemalloc lz4-c zstd -c conda-forge
-   conda activate pyarrow-dev
+    conda create -y -n pyarrow-dev -c conda-forge ^
+        --file arrow\\ci\\conda_env_cpp.yml ^
+        --file arrow\\ci\\conda_env_python.yml ^
```
Comment (wesm) [inline python/doc/source/development.rst]: Hm, I think just single backslashes is OK? I will try it out on Windows before merging

Comment (kszucs) [reply python/doc/source/development.rst]: I'll render the docs, because I put those to escape the single backslash.

Comment (kszucs) [reply python/doc/source/development.rst]: Fixed it

**Review by wesm — APPROVED** [review]
+1. Thanks everyone!

**Comment by wesm** [comment]
Argh so one problem: autoconf is not cross-platform. Does this cause issues on Windows? I can test it out today if no one else has time

**Comment by kszucs** [comment]
@wesm I don't have a windows VM

**Comment by wesm** [comment]
I will try it out on my Windows machine and merge if no issues

**Comment by wesm** [comment]
So we'll have to omit autoconf from conda_env_cpp.yml

```
(pyarrow-dev) λ conda create -n conda-env-test --file=ci\conda_env_cpp.yml python=3.6  
Solving environment: failed                                                            
                                                                                       
PackagesNotFoundError: The following packages are not available from current channels: 
                                                                                       
  - autoconf                                                                           
                                                                                       
Current channels:                                                                      
                                                                                       
  - https://conda.anaconda.org/conda-forge/win-64                                      
  - https://conda.anaconda.org/conda-forge/noarch                                      
  - https://repo.anaconda.com/pkgs/main/win-64                                         
  - https://repo.anaconda.com/pkgs/main/noarch                                         
  - https://repo.anaconda.com/pkgs/free/win-64                                         
  - https://repo.anaconda.com/pkgs/free/noarch                                         
  - https://repo.anaconda.com/pkgs/r/win-64                                            
  - https://repo.anaconda.com/pkgs/r/noarch                                            
  - https://repo.anaconda.com/pkgs/pro/win-64                                          
  - https://repo.anaconda.com/pkgs/pro/noarch                                          
  - https://repo.anaconda.com/pkgs/msys2/win-64                                        
  - https://repo.anaconda.com/pkgs/msys2/noarch                                        
                                                                                       
To search for alternate channels that may provide the conda package you're             
looking for, navigate to                                                               
                                                                                       
    https://anaconda.org                                                               
                                                                                       
and use the search bar at the top of the page.                                         
```

File: cpp/Dockerfile [hunk 3]
```diff
@@ -39,7 +37,8 @@ ADD ci/docker_install_conda.sh \
     /arrow/ci/
 RUN arrow/ci/docker_install_conda.sh && \
     conda install -c conda-forge \
-        --file arrow/ci/conda_env_cpp.yml && \
+        --file arrow/ci/conda_env_cpp.yml \
+        --file arrow/ci/conda_env_unix.yml && \
```
Comment (kszucs) [inline cpp/Dockerfile]: `conda_env_unix.yml` is missing, will resolve in #3059

**Pull request closed without merging**